### PR TITLE
Completly remake merged pull request no. 84

### PR DESCRIPTION
**Pull request by tmszi**
Completly remake merged pull request https://github.com/auto-mat/klub/pull/84

### CODE REVIEW

File: apps/aklub/admin.py [hunk 1]
```diff
@@ -80,6 +84,72 @@ def admin_links(args_generator):
...
+    parent_model_fields = [f.name for f in parent_model._meta.fields]
+    result = {}
+    for child_model in parent_model.__subclasses__():
+        child_model_fields = [f.name for f in child_model._meta.fields]
+        child_model_fields_only = list(
+            set(parent_model_fields).symmetric_difference(set(child_model_fields)),
+        )
+        for field in exclude_fields:
+            child_model_fields_only.remove(field)
+        result[child_model._meta.model_name] = child_model_fields_only
```
Comment (timthelion) [inline apps/aklub/admin.py]: Definuj `exclude_fields` jako set `exclude_fields = {"profile_ptr"}`.

Pak řádky 93-98 se daji psat jako

````
        child_model_fields_only = set(parent_model_fields).symmetric_difference(set(child_model_fields))
	child_model_fields_only = child_model_fields_only - exclude_fields
	result[child_model._meta.model_name] = list(child_model_fields_only)
````

**Comment by timthelion** [comment]
Lokálně migrace mě chazí tohle:
````
(klub-v) test@7dbef627891c:/klub-v$ python3 manage.py migrate
Operations to perform:
  Apply all migrations: admin, admin_tools_stats, advanced_filters, aklub, auth, celery_monitor, contenttypes, dashboard, denorm, django_celery_beat, helpdesk, import_export_celery, menu, post_office, repolinks, sessions, sites, smmapdfs
Running migrations:
  Applying admin.0004_auto_20190826_2034... OK
  Applying auth.0013_auto_20190826_1927... OK
  Applying aklub.0036_auto_20190826_1927... OK
  Applying aklub.0037_create_new_polymorphic_profile_model...Traceback (most recent call last):
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.NotNullViolation: column "profile_ptr_id" contains null values


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 26, in <module>
    execute_from_command_line(sys.argv)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 234, in handle
    fake_initial=fake_initial,
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/operations/fields.py", line 249, in database_forwards
    schema_editor.alter_field(from_model, from_field, to_field)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 535, in alter_field
    old_db_params, new_db_params, strict)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/postgresql/schema.py", line 124, in _alter_field
    new_db_params, strict,
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 685, in _alter_field
    params,
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 137, in execute
    cursor.execute(sql, params)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 99, in execute
    return super().execute(sql, params)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/raven/contrib/django/client.py", line 127, in execute
    return real_execute(self, sql, params)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: column "profile_ptr_id" contains null values

````

File: apps/aklub/admin.py (hunk 1 above)
Comment (tmszi) [reply apps/aklub/admin.py]: Vďaka, opravím.

**Comment by timthelion** [comment]
userprofile

Už mě to prošlo. Nemůžu ale najit v menu "User profiles" a "Company Profiles" ačkoliv ty tabulky existuji. Registroval jsi je v adminu?

**Comment by tmszi** [comment]
> userprofile
> 
> Už mě to prošlo. Nemůžu ale najit v menu "User profiles" a "Company Profiles" ačkoliv ty tabulky existuji. Registroval jsi je v adminu?

Zaregistrované sú oba child modely  (`'User/CompanyProfile'`), default som ich schoval, zmeň atribút `show_in_index=True` v oboch admin modeloch. `Profile` model je polymorfický model, v admin rozhraní  vo view liste tohto modelu nájdeš oba child typy modelov, teda `'User/CompanyProfile'` model.

**Comment by tmszi** [comment]
> Lokálně migrace mě chazí tohle:
> 
> ```
> (klub-v) test@7dbef627891c:/klub-v$ python3 manage.py migrate
> Operations to perform:
>   Apply all migrations: admin, admin_tools_stats, advanced_filters, aklub, auth, celery_monitor, contenttypes, dashboard, denorm, django_celery_beat, helpdesk, import_export_celery, menu, post_office, repolinks, sessions, sites, smmapdfs
> Running migrations:
>   Applying admin.0004_auto_20190826_2034... OK
>   Applying auth.0013_auto_20190826_1927... OK
>   Applying aklub.0036_auto_20190826_1927... OK
>   Applying aklub.0037_create_new_polymorphic_profile_model...Traceback (most recent call last):
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
>     return self.cursor.execute(sql, params)
> psycopg2.errors.NotNullViolation: column "profile_ptr_id" contains null values
> 
> 
> The above exception was the direct cause of the following exception:
> 
> Traceback (most recent call last):
>   File "manage.py", line 26, in <module>
>     execute_from_command_line(sys.argv)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
>     utility.execute()
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
>     self.fetch_command(subcommand).run_from_argv(self.argv)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
>     self.execute(*args, **cmd_options)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
>     output = self.handle(*args, **options)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
>     res = handle_func(*args, **kwargs)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 234, in handle
>     fake_initial=fake_initial,
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/executor.py", line 117, in migrate
>     state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
>     state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
>     state = migration.apply(state, schema_editor)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/migration.py", line 124, in apply
>     operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/migrations/operations/fields.py", line 249, in database_forwards
>     schema_editor.alter_field(from_model, from_field, to_field)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 535, in alter_field
>     old_db_params, new_db_params, strict)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/postgresql/schema.py", line 124, in _alter_field
>     new_db_params, strict,
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 685, in _alter_field
>     params,
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/base/schema.py", line 137, in execute
>     cursor.execute(sql, params)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 99, in execute
>     return super().execute(sql, params)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/raven/contrib/django/client.py", line 127, in execute
>     return real_execute(self, sql, params)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 67, in execute
>     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
>     return executor(sql, params, many, context)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
>     return self.cursor.execute(sql, params)
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
>     raise dj_exc_value.with_traceback(traceback) from exc_value
>   File "/home/test/.local/share/virtualenvs/klub-v-nnivlwsY/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
>     return self.cursor.execute(sql, params)
> django.db.utils.IntegrityError: column "profile_ptr_id" contains null values
> ```

Tú chybu som vyriešil , je to spôsobené tým že pri migrácii sa migrujú aj data, ak existujú zo starého 'UserProfile' modelu do nového polymorfického 'Profile' modelu resp do jeho child modelov 'User/CompanyProfile'.

**Comment by timthelion** [comment]
Jestli dáš `pipenv install --dev` tak máš tam v pipenvu i `flake8`. Není nutné po každé čekat na rozběhnutí travisu kvůli par malých změn.

**Comment by tmszi** [comment]
To samozrejme viem aj používam  aj v Emacse mám tiež povolený flake8  
checker...
Len v travis mi hlásil flake8 error, a na localhoste som to mal valídne.

Citát Timothy Hobbs <notifications@github.com>:

**Comment by timthelion** [comment]
To je protože se použiva flake8-strict což je zřejmě trochu jiné... Je
to popsané v .travis.yml

**Comment by tmszi** [comment]
Áno ten používam :-), config travisu poznám.

Tak zatiaľ,

Toms

Citát Timothy Hobbs <notifications@github.com>:

**Pull request merged**